internal/wasm/memory.go

```diff
@@ -57,12 +58,22 @@ type MemoryInstance struct {
 	// waiters implements atomic wait and notify. It is implemented similarly to golang.org/x/sync/semaphore,
 	// with a fixed weight of 1 and no spurious notifications.
 	waiters sync.Map
+
+	allocator experimental.MemoryBuffer
```

Comment:
This is not `allocator`, but a buffer, which sounds confusing to me. I would suggest to rename it to `expBuffer`

```suggestion
	expBuffer experimental.MemoryBuffer
```